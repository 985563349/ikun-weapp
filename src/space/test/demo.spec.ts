import path from 'node:path';
import simulate from 'miniprogram-simulate';

test('loading snapshot', async () => {
  const id = simulate.load(path.resolve(__dirname, '../demo/space'), 'basic');
  const container = simulate.render(id);

  container.attach(document.createElement('parent-wrapper'));
  expect(container.toJSON()).toMatchSnapshot();
});
