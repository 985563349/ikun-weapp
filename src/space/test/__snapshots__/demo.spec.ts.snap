// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading snapshot 1`] = `
<basic>
  <wx-view>
    <wx-view>
      <wx-view>
        基本用法
      </wx-view>
      <wx-view>
        <k-space>
          <k-button>
            Default
          </k-button>
          <k-button
            color="primary"
          >
            Primary
          </k-button>
          <k-button
            color="success"
          >
            Success
          </k-button>
        </k-space>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        垂直间距
      </wx-view>
      <wx-view>
        <k-space
          style="display: flex"
          vertical="{{true}}"
        >
          <k-button
            expand="block"
          >
            Default
          </k-button>
          <k-button
            color="primary"
            expand="block"
          >
            Primary
          </k-button>
          <k-button
            color="success"
            expand="block"
          >
            Success
          </k-button>
        </k-space>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        间距大小
      </wx-view>
      <wx-view>
        <k-space
          size="md"
        >
          <k-button>
            Default
          </k-button>
          <k-button
            color="primary"
          >
            Primary
          </k-button>
          <k-button
            color="success"
          >
            Success
          </k-button>
        </k-space>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        对齐
      </wx-view>
      <wx-view>
        <k-space
          align="center"
        >
          <wx-text>
            center
          </wx-text>
          <k-button
            color="primary"
            size="lg"
          >
            Primary
          </k-button>
          <wx-view
            style="padding: 40rpx 20rpx 60rpx; background-color: #d4d4d8"
          >
            Block
          </wx-view>
        </k-space>
      </wx-view>
    </wx-view>
    <wx-view>
      <wx-view>
        自动换行
      </wx-view>
      <wx-view>
        <k-space
          wrap="{{true}}"
        >
          <k-button>
            Default
          </k-button>
          <k-button
            color="primary"
          >
            Primary
          </k-button>
          <k-button
            color="success"
          >
            Success
          </k-button>
          <k-button
            color="warning"
          >
            Warning
          </k-button>
          <k-button
            color="danger"
          >
            Danger
          </k-button>
        </k-space>
      </wx-view>
    </wx-view>
  </wx-view>
</basic>
`;
